src/utils.cpp

```diff
@@ -714,7 +714,11 @@ namespace {
   std::string unpack_kheaders_tar_xz(const struct utsname& utsname)
   {
     std::error_code ec;
+#if defined(__ANDROID__)
+    std_filesystem::path path_prefix{ "/data/local/tmp" };
+#else
     std_filesystem::path path_prefix{ "/tmp" };
+#endif
     std_filesystem::path path_kheaders{ "/sys/kernel/kheaders.tar.xz" };
     if (const char* tmpdir = ::getenv("TMPDIR")) {
```

Comment:
The above seems reasonable, but I'm wondering why `TMPDIR` is not set. Or if it is, what it's set to on android. Wikipedia says it's a posix standard: https://en.wikipedia.org/wiki/TMPDIR

Reply:
afaik, Android doesn't have global tmp dir due to security considerations. Instead global tmp dir, the apps are able to access the [app-specific cache dir](https://developer.android.com/reference/android/content/Context#getCacheDir()). (/data/data/${APP_ID}/cache or /data/user/0/$APP_ID}/cache). 
For the shell or testing frame, Android gives this `/data/local/tmp` as `TMPDIR`